archunit/src/main/java/com/tngtech/archunit/lang/conditions/ArchConditions.java

```diff
@@ -1321,33 +1326,68 @@ public void check(JavaClass javaClass, ConditionEvents events) {
         }
     }
 
-    private static class NumberOfElementsCondition extends ArchCondition<JavaClass> {
+    private static abstract class NumberOfElementsCondition<T> extends ArchCondition<T> {
         private final DescribedPredicate<Integer> predicate;
-        private final SortedSet<String> allClassNames = new TreeSet<>();
+        private final SortedSet<String> allElementNames = new TreeSet<>();
 
         NumberOfElementsCondition(DescribedPredicate<? super Integer> predicate) {
             super("contain number of elements " + predicate.getDescription());
             this.predicate = predicate.forSubtype();
         }
 
         @Override
-        public void check(JavaClass item, ConditionEvents events) {
-            allClassNames.add(item.getName());
+        public void check(T item, ConditionEvents events) {
+            allElementNames.add(getUniqueName(item));
         }
 
+        protected abstract String getUniqueName(T item);
+
         @Override
         public void finish(ConditionEvents events) {
-            int size = allClassNames.size();
+            int size = allElementNames.size();
             boolean conditionSatisfied = predicate.apply(size);
-            String message = String.format("there is/are %d element(s) in classes %s", size, join(allClassNames));
-            events.add(new SimpleConditionEvent(size, conditionSatisfied, message));
+            events.add(new SimpleConditionEvent(size, conditionSatisfied, getMessage(size, join(allElementNames))));
         }
 
+        protected abstract String getMessage(int size, String elements);
+
         private String join(SortedSet<String> strings) {
             return "[" + Joiner.on(", ").join(strings) + "]";
         }
     }
 
+    private static class NumberOfClassesCondition extends NumberOfElementsCondition<JavaClass> {
+        NumberOfClassesCondition(DescribedPredicate<? super Integer> predicate) {
+            super(predicate);
+        }
+
+        @Override
+        protected String getUniqueName(JavaClass item) {
+            return item.getName();
```

Comment:
You can also use `JavaClass.getFullName`, it's the same. GIven that, it should be enough to parameterize "classes" vs "members", which should make the subclasses obsolete, don't you think?